### DISCUSSION TITLE
Does NUnit have a suggested approach for test configuration compatible with IConfiguration?

### DISCUSSION THREAD

**Discussion by madelson** [nunit/nunit#4367] [Q&A] [answered]
We have a few use-cases for providing configuration values to our test projects. Examples include:
* Database credentials
* Paths to external executable dependencies
* Flag to indicate environment

In .NET Framework, we would transform these values in the test's app.config file and then read from `ConfigurationManager.AppSettings` . In the .NET Core world, many things leverage `IConfiguration` instead.

Is there a recommended way to flow configuration values to NUnit tests? I found [this page](https://docs.nunit.org/articles/nunit/technical-notes/usage/Configuration-Files.html) but that refers to .config files.

We could of course load an `IConfiguration` instance ourselves [as suggested here](https://stackoverflow.com/questions/39791634/read-appsettings-json-values-in-net-core-test-project) but I was curious if there was a way to do this that would be "idiomatic" for NUnit.

**Comment by OsirisTerje** [comment, marked as answer]
Currently no, and I am not sure if IConfiguration is - what to call it - "fundamental" enough for us to take a dependency on.  We try to avoid taking dependencies on other packages.  
The NUnit config values you refer to are for the NUnit.project types, and those are only used in the NUnit.Console.
If you're using dotnet and/or visual studio, the .runsettings file is the way to push items to the test system.  
You can add properties to the [TestRunParameters](https://learn.microsoft.com/en-us/visualstudio/test/configure-unit-tests-by-using-a-dot-runsettings-file?view=vs-2022#testrunparameters) property bag, and these can be read by the NUnit TestContext, see https://stackoverflow.com/questions/75062954/i-have-multiple-testrunparameters-in-an-nunit-runsettings-file-how-can-i-loop
